cli/azd/pkg/tools/azcli/resources.go

```diff
@@ -0,0 +1,193 @@
+package azcli
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/azure/azure-dev/cli/azd/pkg/httputil"
+	"github.com/azure/azure-dev/cli/azd/pkg/identity"
+)
+
+func (cli *azCli) GetResource(ctx context.Context, subscriptionId string, resourceId string) (AzCliResourceExtended, error) {
+	client, err := cli.createResourcesClient(ctx, subscriptionId)
+	if err != nil {
+		return AzCliResourceExtended{}, err
+	}
+
+	res, err := client.GetByID(ctx, resourceId, "", nil)
+	if err != nil {
+		return AzCliResourceExtended{}, fmt.Errorf("getting resource by id: %w", err)
+	}
+
+	return AzCliResourceExtended{
+		AzCliResource: AzCliResource{
+			Id:       *res.ID,
+			Name:     *res.Name,
+			Type:     *res.Type,
+			Location: *res.Location,
+		},
+		Kind: *res.Kind,
+	}, nil
+}
+
+func (cli *azCli) ListResourceGroupResources(ctx context.Context, subscriptionId string, resourceGroupName string, listOptions *ListResourceGroupResourcesOptions) ([]AzCliResource, error) {
+	client, err := cli.createResourcesClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter expression on the underlying REST API are different from --query param in az cli.
+	// https://learn.microsoft.com/en-us/rest/api/resources/resources/list-by-resource-group#uri-parameters
```

Comment:
Maybe add this link as documentation to the `Filter` parameter on the options struct as well (or just move it to there, so it's more "public")?